### PR TITLE
Initial commit of common

### PR DESCRIPTION
**Pull request by rlazo**
Changes are the minimum necessary to extract the common functionality into the new sdk, mostly renaming packages, removing internal declarations, and that's it. Follow up CLs will add tests, remove data classes, and improve exception reporting.

### CODE REVIEW

File: common/gradle.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+version=0.2.0
```
Comment (daymxn) [inline common/gradle.properties:1]: Why `0.2.0`?

**Review by daymxn — APPROVED** [review]
Left a comment, but LGTM!

Also...
> Follow up CLs will add tests, remove data classes, and improve exception reporting.

...

> remove data classes

nooo

**Review by davidmotson — APPROVED** [review]
LGTM, but ktfmt

File: common/gradle.properties (hunk 1 above)
Comment (rlazo) [reply common/gradle.properties:1]: fat finger :-)

**Pull request merged**